var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -323,16 +322,12 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on('adios2', when='+adios2')
     depends_on('blas')
-    depends_on('boost', when='+boost')
+    depends_on('boost+graph+math+exception+stacktrace', when='+boost')
     # Need to revisit the requirement of STK
-    depends_on('boost', when='+stk')
+    depends_on('boost+graph+math+exception+stacktrace', when='+stk')
```

Comment:
Maybe this was created before the concretizer, but I think this should probably be `conflicts('~boost', when='+stk')`, although who knows whether the new concretizer can deduce this correctly... Otherwise `~boost` can still get a copy of `boost` built in. But leaving it as-is certainly is safer...